source/common/common/cleanup.h

```diff
@@ -14,4 +15,18 @@ class Cleanup {
   std::function<void()> f_;
 };
 
+// RAII helper class to add an element to an std::list on construction and erase
+// it on destruction.
+template <class T> class ListAddAndRemove {
```

Comment:
Can we call this `RaiiListElement` or something like that, to make sure it's clear that we're dealing with a list element that is self cleaning?

Reply:
Done